### PR TITLE
feat: display sns projects status pending with time window, open and complete + sort

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

When a Sns proposal is accepted, the decentralized sale can effectively begin later on that's why we already want to display such projects in the "opportunities" section. Likewise, we want to display these projects for a while in the same section if they complete.

Finally, to ensure everything is sorted always in a same way, we want to sort the sns projects according swap start date.

# Changes

- rename `openProjectsStore` to `activeProjects` store and apply new filter "pending + time window or open or complete"
- move store filter to a new `projects.utils` file
- sort sns concatenated projects according swap start date

### CODE REVIEW

File: frontend/src/lib/components/launchpad/Projects.svelte [hunk 1]
```diff
@@ -5,7 +5,7 @@
   } from "../../services/sns.services";
   import { i18n } from "../../stores/i18n";
   import {
-    openProjectsStore,
+    activePadProjectsStore,
```
Comment (mstrasinskis) [inline frontend/src/lib/components/launchpad/Projects.svelte:8]: Out of curiosity, why not `launchpadProjectsStore`?

**Review by mstrasinskis — APPROVED** [review]
LGTM, thank you!

File: frontend/src/lib/components/launchpad/Projects.svelte (hunk 1 above)
Comment (peterpeterparker) [reply frontend/src/lib/components/launchpad/Projects.svelte:8]: That was the first name but then I thought it make it really tight to screen name particularly if we ever used it elsewhere, that's why ultimately I thought about a more generic name. Make sense or you rather like launchPad...?

File: frontend/src/lib/utils/sns.utils.ts [hunk 2]
```diff
@@ -38,15 +63,17 @@ export const concatSnsSummaries = ([summaries, swaps]: [
...
-      init: fromNullable(swap.init) as SnsSwapInit,
-      state: fromNullable(swap.state) as SnsSwapState,
-    },
-  }));
+  return sortSnsSummaries(
+    validSwapSummaries.map(({ swap, ...rest }) => ({
+      ...rest,
+      swap: {
+        // We know for sure that init and state are defined because we check in previous filter that there are not undefined
+        // TODO: There might be a cleaner way than a type cast to make TypeScript checks these are defined
```
Comment (mstrasinskis) [inline frontend/src/lib/utils/sns.utils.ts:71]: How about having separate function for this? 
e.g. `export const fromDefinedNullable = <T>(value: [T]): T =>  value[0];`

File: frontend/src/lib/components/launchpad/Projects.svelte (hunk 1 above)
Comment (mstrasinskis) [reply frontend/src/lib/components/launchpad/Projects.svelte:8]: I don't have any strong opinion. So let's leave it ;)

Comment (mstrasinskis) [reply frontend/src/lib/components/launchpad/Projects.svelte:8]: Thank you for the explanation.

File: frontend/src/lib/utils/sns.utils.ts (hunk 2 above)
Comment (peterpeterparker) [reply frontend/src/lib/utils/sns.utils.ts:71]: Like it. Can you add it to the new utils lib?

**Pull request merged**